Makefile

```diff
@@ -21,13 +21,14 @@ INSTALL_DATA= $(INSTALL) -m 0644
 
 # C compilation flags
 CFLAGS   = -DPT_DEBUG -O2 -std=c99 -pedantic -Wall -Wextra -Wformat-security
-CPPFLAGS = -I$(LUA_INCDIR) -I.
+# Explicitly mention which Lua headers to capture
+CPPFLAGS = -isystem$(LUA_INCDIR) -I.
```

Comment:
Why doesn't the -I work? This -isystem is very strange because it's *not* a system headerfile.

Reply:
This is so that compiler treats the Lua include dir as a system dir (increase the resolving priority). Say, for example, we are building against Lua internals, which has it's headers files in `/usr/local/include` directory. Now, if we already have Lua installed system-wide, the compiler would resolve the header files in `/usr/include` rather than `/usr/local/include`, because system headers (specifically /usr/include) are resolved first, then /usr/local/include headers, then headers specified by the `-I` flag and finally current directory (If included using quotes).

Problem here is, when you have different version of Lua installed in `/usr/local`, compiler would include the Lua headers from `/usr/include` instead of `/usr/local/include` because of the priority. The `-isystem` flag bumps up the priority and what I can tell, it bumps up the priority to the maximum.

Besides, everything in the `/usr` directory is part of **SYSTEM**, including `/usr/local` because binaries installed in `/usr` or `/usr/local/` can be used by all the users in system. `/usr/local` is just used by softwares which does not use standard package manager.

Reply:
https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html

According to the gcc manual, -I should have higher priority than -isystem

> You can use -I to override a system header file, substituting your own version, since these directories are searched before the standard system header file directories. However, you should not use this option to add directories that contain vendor-supplied system header files; use -isystem for that.

/usr/local is not "vendor supplied"

Reply:
Why are you compiling with PREFIX=/usr?

Should be PREFIX=/usr/local

Reply:
Not the point actually. Point is our desired headers are only properly resolved when we use `-isystem` instead of `-I`.